espnet2/train/dataset.py

```diff
@@ -64,10 +66,11 @@ def __getitem__(self, key: str) -> np.ndarray:
                     f"Unexpected type: {type(retval[0])}, {type(retval[1])}"
                 )
 
-            if self.rate is not None and self.rate != rate:
-                raise RuntimeError(
-                    f"Sampling rates are mismatched: {self.rate} != {rate}"
-                )
+            if not self.allow_multi_rates:
+                if self.rate is not None and self.rate != rate:
```

Comment:
How about reducing two `if`  into one？
`if not self.allow_multi_rates and self.rate is not None and self.rate != rate:`

Reply:
OK